fastlane_bot/events/exchanges/uniswap_v3.py

```diff
@@ -42,6 +47,9 @@ def get_events(self, contract: Contract) -> List[Type[Contract]]:
     async def get_fee(self, address: str, contract: Contract) -> Tuple[str, float]:
         fee = await contract.functions.fee().call()
         fee_float = float(fee) / 1e6
+        if self.exchange_name == NILE_V3_NAME:
+            fee_float = await contract.functions.currentFee().call()
```

Comment:
Why function `currentFee` and not function `fee`?
For UniV3 pools, we are using function `fee`.
Might be worth verifying exactly which function is required here.
You can do that by looking into the source code of this pool contract, checking which fee is used in the swap function, and then following it to see which of these two functions returns that specific fee.

Reply:
Also, there is something really wrong here, as in one case you are dividing by the resolution, and in one case you don't.
So `fee_float` is sometimes returned after being divided by `16e`, and sometimes being returned as read from the contract.
How could that possibly work correctly further down the road, when the caller of this function receives this returned value and needs to processes it?
Also note that you seem to have made a similar change in file `run_blockchain_terraformer.py`, where you actually do divide by `1000000` in both cases.
So even without taking the above into consideration, it is pretty clear that at least one of these two changes is incorrect.